invenio_app_ils/indexer.py

```diff
@@ -19,20 +19,19 @@
 
 from invenio_app_ils.circulation.utils import circulation_document_retriever, \
     circulation_items_retriever
-from invenio_app_ils.records.api import Document, Item
+from invenio_app_ils.records.api import Document, InternalLocation, Item, \
+    Location
+from invenio_app_ils.search.api import InternalLocationSearch
 
 
 @shared_task(ignore_result=True)
 def index_loans_after_item_indexed(item_pid):
     """Index loan to refresh item reference."""
     loan_search = search_by_pid(item_pid=item_pid)
-    loan_ids = []
     for loan in loan_search.scan():
         record = Loan.get_record_by_pid(loan[Loan.pid_field])
         if record:
-            loan_ids.append(record.id)
-
-    RecordIndexer().bulk_index(loan_ids)
+            RecordIndexer().index(record)
```

Comment:
Why not indexing asynchronous here?

Reply:
According to @ntarocco, we don't want to wait for the 5 minutes for the task to be picked up. But this is just our idea, feel free to elaborate.

Reply:
where is defined the 5 minutes? I assume you mean the indexing task?

Reply:
`bulk_index` is adding to the queue, then we need to `process` the queue or wait for 5 minutes.
See: https://github.com/inveniosoftware/invenio-app-ils/blob/master/invenio_app_ils/cli.py#L323

@zzacharo WDYT?
In this way, it is already async with `@shared_task`

Reply:
I mean is already asynchronous the task so this assumes that we don't really waiting for the result either way. But I see the point of not waiting, not sure if is a problem. I think that the ES is more performant in the bulk index form but I am not sure though. We can change it and see if it cause troubles or not in the future.

Reply:
Agree, it just that the process_queue triggered an error when we tried (we did not investigate).
@topless do you want to try that? It is a little performance gain.